spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -287,4 +302,13 @@ private T getSessionCatalog() {
         "Please make sure your are replacing Spark's default catalog, named 'spark_catalog'.");
     return sessionCatalog;
   }
+
+  private boolean callPurgeTable(Object catalog, Identifier ident) {
```

Comment:
I don't think it's a good idea to use reflection here. I'd probably wait to merge this until after the Spark refactor so that we can implement new APIs from 3.1

Reply:
+1

Reply:
OK, waiting for the Spark refactor.